ledger/ledger_test.go

```diff
@@ -887,3 +887,33 @@ func TestLedgerBlockHdrCaching(t *testing.T) {
 		require.Equal(t, blk.BlockHeader, hdr)
 	}
 }
+
+func TestLedgerReload(t *testing.T) {
+	dbName := fmt.Sprintf("%s.%d", t.Name(), crypto.RandUint64())
+	genesisInitState := getInitState()
+	const inMem = true
+	cfg := config.GetDefaultLocal()
+	cfg.Archival = true
+	log := logging.TestingLog(t)
+	l, err := OpenLedger(log, dbName, inMem, genesisInitState, cfg)
+	require.NoError(t, err)
+	defer l.Close()
+
+	blk := genesisInitState.Block
+	for i := 0; i < 128; i++ {
+		blk.BlockHeader.Round++
+		blk.BlockHeader.TimeStamp += int64(crypto.RandUint64() % 100 * 1000)
+		l.AddBlock(blk, agreement.Certificate{})
+
+		if i%7 == 0 {
+			l.reloadLedger()
```

Comment:
why the error is not checked here?

Reply:
you're right; I should have tested the error code. Could you please review the changes?